naming/dns_resolver_test.go

```diff
@@ -0,0 +1,304 @@
+package naming
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+)
+
+func newUpdate(op Operation, addr string) *Update {
+	return &Update{
+		Op:   op,
+		Addr: addr,
+	}
+}
+
+func newUpdateWithMD(op Operation, addr, lb string) *Update {
+	return &Update{
+		Op:       op,
+		Addr:     addr,
+		Metadata: AddrMetadataGRPCLB{AddrType: GRPCLB, ServerName: lb},
+	}
+}
+
+func newSRVRR(target string, port uint16) *net.SRV {
+	return &net.SRV{
+		Target: target,
+		Port:   port,
+	}
+}
+
+var updateTc = []struct {
```

Comment:
It's more customary to declare this inside the test that uses it, since it's used only once.